### PR TITLE
Fix source hierarchy

### PR DESCRIPTION
**Pull request by panglesd**
The recent refactor of source locations into identifiers #969 introduced some changes in how source trees are organized and displayed. The changes were not caught, since the test was not run, due to a too restrictive constraint on the ocaml version.
This PR:
- enables source tests on all `4.14.*` versions
- Consider again source pages as "leaf pages" (name them `filename.ml.html` instead of `filename.ml/index.html`)
- Display full path in directory title: Goes from 
image
back to 
image
(Was the change on purpose? The full path is a little bit redundant with the breadcrumb, but more visible)

### CODE REVIEW

**Comment by jonludlam** [comment]
The change was not on purpose! Thanks for sorting this.

**Comment by panglesd** [comment]
> Why is there an upper bound on the ocaml version?

The source hierarchy is using some internals from the compiler, so the test output is not very stable between OCaml version...

**Review by Julow — APPROVED** [review]
Looks good !

There's an other broken version constraint in `test/integration/dune` while you are at it.

File: test/sources/dune [hunk 1]
```diff
@@ -7,5 +7,7 @@
 
 (cram
  (enabled_if
-  (= %{ocaml_version} 4.14.0))
+  (and
+   (>= %{ocaml_version} 4.14.0)
+   (< %{ocaml_version} 5.0.0)))
```
[the remark this answers is not in the document]
Comment (jonludlam) [reply test/sources/dune:12]: I'd rather leave this as-is for now and fix it properly by making the anchors stable :-)

**Comment by jonludlam** [comment]
Thanks @panglesd !

**Pull request merged**